src/DBFTables.jl

```diff
@@ -1,150 +1,293 @@
 module DBFTables
 
-using DataFrames, Printf
-
-# Read DBF files in xBase format
-# Files written in this format have the extension .dbf
-# Implemented: dBase III+ (w/o memo)
-
-struct DBFFieldDescriptor
-	nam::String
-	typ::Type
-	len::Int8
-	dec::Int8
-end
-
-struct DBFHeader
-	version::UInt8
-	lastUpdate::String
-	records::Int32
-	hsize::Int16
-	rsize::Int16
-	incomplete::Bool
-	encrypted::Bool
-	mdx::Bool
-	langId::UInt8
-	fields::Vector{DBFFieldDescriptor}
-end
-
-function dbf_field_type(fld::Char, dec::UInt8)
-	rt = Nothing
-	if fld == 'C'
-		rt = String
-	elseif fld == 'D'
-		rt = String
-	elseif fld == 'N'
-		if dec > 0
-			rt = Float64
-		else
-			rt = Int
-		end
-	elseif fld == 'F' || fld == 'O'
-		rt = Float64
-	elseif fld == 'I' || fld == '+'
-		rt = Integer
-	elseif fld == 'L'
-		rt = Bool
-	else
-		 warn("Unknown record type: $(fld)")
-	end
-	return rt
+using Printf, Tables, WeakRefStrings
+
+"Field/column descriptor, part of the Header"
+struct FieldDescriptor
+    name::Symbol
+    type::DataType
```

Comment:
Usually, this would be `Type` if it can ever be a `Union` type, because:
```julia
julia> Union{Int, Missing} isa DataType
false

julia> Union{Int, Missing} isa Type
true
```
is the `type` field here guaranteed to be a specific set of types at all?